cpp/src/copying/contiguous_split.cu

```diff
@@ -164,6 +404,69 @@ struct column_copy_functor {
    }
 };
 
+/**
+ * @brief Information about a string column in a table view.
+ * 
+ * Used internally by preprocess_string_column_info as part of a device-accessible 
+ * vector for computing final string information in a single kernel call.
+ */
+struct column_preprocess_info {
+   size_type                  index;
+   size_type                  offset;
+   size_type                  size;
+   bool                       nullable;      
+   cudf::column_device_view   offsets;
+};
+
+/**
+ * @brief Preprocess information about all strings columns in a table view.
+ * 
+ * In order to minimize how often we touch the cpu, we need to preprocess various pieces of information
+ * about the string columns in a table as a batch process.  This function builds a list of the offset
+ * columns for all input string columns and computes this information with a single thrust call.  In addition,
+ * the vector returned is allocated for -all- columns in the table so further processing of non-string columns
+ * can happen afterwards.
+ * 
+ * The key things this function avoids
+ * - avoiding reaching into gpu memory on the cpu to retrieve offsets to compute string sizes.
+ * - creating column_device_views on the base string_column_view itself as that causes gpu memory allocation.
+ */
+thrust::host_vector<column_split_info> preprocess_string_column_info(cudf::table_view const& t, thrust::device_vector<column_split_info>& device_split_info, cudaStream_t stream)
+{          
+   // build a list of all the offset columns and their indices for all input string columns and put them on the gpu   
+   thrust::host_vector<column_preprocess_info> offset_columns;
+   offset_columns.reserve(t.num_columns());  // worst case
+   size_type column_index = 0;
+   std::for_each(t.begin(), t.end(), [&offset_columns, &column_index](cudf::column_view const& c){
+      if(c.type().id() == STRING){
```

Comment:
Right.  I skipped the type dispatch here because we only ever do work for the string case.  Everything else is a nop.  So it seemed like adding a functor and a type dispatcher call would add unnecessary obfuscation - you'd have to jump around the code more to see what was actually happening than just having the simple if statement inline.